src/runtime/entries/cloudflare.ts

```diff
@@ -3,27 +3,38 @@ import {
   getAssetFromKV,
   mapRequestToAsset,
 } from "@cloudflare/kv-asset-handler";
-import { withoutBase } from "ufo";
+import { withLeadingSlash, withoutBase, withoutTrailingSlash } from "ufo";
+import { splitCookiesString } from "h3";
 import { requestHasBody } from "../utils";
 import { nitroApp } from "#internal/nitro/app";
 import { useRuntimeConfig } from "#internal/nitro";
-import { getPublicAssetMeta } from "#internal/nitro/virtual/public-assets";
+import {
+  getPublicAssetMeta,
+  isPublicAssetURL,
+} from "#internal/nitro/virtual/public-assets";
 
 addEventListener("fetch", (event: any) => {
   event.respondWith(handleEvent(event));
 });
 
 async function handleEvent(event: FetchEvent) {
-  try {
-    return await getAssetFromKV(event, {
-      cacheControl: assetsCacheControl,
-      mapRequestToAsset: baseURLModifier,
-    });
-  } catch {
-    // Ignore
+  const url = new URL(event.request.url);
+  const id = decodeURIComponent(
+    withLeadingSlash(withoutTrailingSlash(url.pathname))
+  );
+
+  // Fetch public assets from KV only
+  if (isPublicAssetURL(id)) {
```

Comment:
I agree this needs to happen, but it should be handled in a different PR (https://github.com/unjs/nitro/pull/1236/files)